File: task-manager-ui/src/app/views/base/tasks/dto/task-request-dto.ts
export interface TaskRequestDto {
  title: string;
  donorId: number;
  description?: string;
  assignedPartnerId?: number;
  validatedBudget?: number;
  requestReceivedAt?: string; // ISO 8601 format
  acceptedAt?: string; // ISO 8601 format
  deadline?: string; // ISO 8601 format
  allocateNotes?: string;
  acceptanceNotes?: string;
  rejectionNotes?: string;
}

File: task-manager-backend/src/main/java/com/task/manage/task/dto/TaskRequestDto.java
package com.task.manage.task.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record TaskRequestDto(
        String title,

        Long donorId,

        String description,

        Long assignedPartnerId,

        BigDecimal validatedBudget,

        LocalDateTime requestReceivedAt,

        LocalDateTime acceptedAt,

        LocalDateTime deadline,

        String allocateNotes,

        String acceptanceNotes,

        String rejectionNotes
) {
}
